src/Microsoft.AspNetCore.Mvc.RazorPages/PageResult.cs

```diff
@@ -56,14 +42,14 @@ public PageResult(PageBase page, object model)
         /// <inheritdoc />
         public override Task ExecuteResultAsync(ActionContext context)
         {
-            if (!object.ReferenceEquals(context, Page.PageContext))
+            if (!(context is PageContext pageContext))
```

Comment:
Is the exception message good enough for people trying to return this result from a controller?